auth/internal/trustboundary/trust_boundary.go

```diff
@@ -0,0 +1,324 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trustboundary
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"cloud.google.com/go/auth/internal"
+)
+
+const (
+	// NoOpEncodedLocations is a special value indicating that no trust boundary is enforced.
+	NoOpEncodedLocations = "0x0"
+	// serviceAccountAllowedLocationsEndpoint is the URL for fetching allowed locations for a given service account email.
+	serviceAccountAllowedLocationsEndpoint = "https://iamcredentials.%s/v1/projects/-/serviceAccounts/%s/allowedLocations"
+)
+
+// DataProvider provides an interface for fetching trust boundary data.
+// It's responsible for obtaining trust boundary information, including caching and specific logic for different credential types.
+type DataProvider interface {
+	// GetTrustBoundaryData retrieves the trust boundary data (type Data).
+	// The accessToken is the bearer token used to authenticate the lookup request to the Trust Boundary API.
+	// The context provided should be used for any network requests.
+	GetTrustBoundaryData(ctx context.Context, accessToken string) (*Data, error)
+}
+
+// ConfigProvider provides specific configuration for trust boundary lookups.
+type ConfigProvider interface {
+	// GetTrustBoundaryEndpoint returns the endpoint URL for the trust boundary lookup.
+	GetTrustBoundaryEndpoint(ctx context.Context) (url string, err error)
+	// GetUniverseDomain returns the universe domain associated with the credential.
+	// It may return an error if the universe domain cannot be determined.
+	GetUniverseDomain(ctx context.Context) (string, error)
+}
+
+// AllowedLocationsResponse is the structure of the response from the Trust Boundary API.
+type AllowedLocationsResponse struct {
+	// Locations is the list of allowed locations.
+	Locations []string `json:"locations"`
+	// EncodedLocations is the encoded representation of the allowed locations.
+	EncodedLocations string `json:"encodedLocations"`
+}
+
+// Data represents the trust boundary data.
+type Data struct {
+	// Locations is the list of locations that the token is allowed to be used in.
+	locations []string
```

Comment:
Any reasons to not just make these public and lose the accessors? Would allow you to remove a decent bit of code.

Reply:
Made them public and removed the accessors.